### PR TITLE
Changes to MatchMerging as its tests by nils.

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
Contains the changes to MatchMerging by Nils. I also refactored the code in some places where my IDE suggested it.

### CODE REVIEW

**Comment by uuqjz** [comment]
@TwoOfTwelve tag me as reviewer

**Comment by tsaglam** [comment]
> @TwoOfTwelve tag me as reviewer

You can do that yourself in the top right corner... ;)

**Comment by uuqjz** [comment]
@tsaglam I cant do that, it seems i am neither mainainer nor StudDev

**Comment by tsaglam** [comment]
Should work now.

File: core/src/test/java/de/jplag/merging/MergingTest.java [hunk 1]
```diff
@@ -194,4 +197,35 @@ void testCorrectMerges() {
...
+        expectedBefore.add(new Match(5, 3, 6));
+        expectedBefore.add(new Match(11, 12, 6));
+        expectedBefore.add(new Match(0, 0, 3));
+        expectedBefore.add(new Match(3, 18, 2));
+        expectedBefore.add(new Match(17, 20, 2));
+
+        List<Match> expectedAfter = new ArrayList<>();
+        expectedAfter.add(new Match(5, 3, 12));
+
+        assertTrue(matchesBefore.equals(expectedBefore) && matchesAfter.equals(expectedAfter));
```
Comment (uuqjz) [inline core/src/test/java/de/jplag/merging/MergingTest.java:229]: Oversight of mine:
Split into two assertEquals please

**Review by uuqjz — CHANGES_REQUESTED** [review]
Comment above

**Review by uuqjz — APPROVED** [review]

**Review by tsaglam — APPROVED** [review]

**Pull request merged**